Lab5

yinyangcolby opened the pull request:
**Hello. I apologize for not having lab 5 just ready yet, so I've created a new branch based off of lab 4 so you can have a link to at least turn in for today. I plan to finish up lab 5 by tomorrow evening, if you don't mind coming back tomorrow to then edit your code review.**

yinyangcolby committed:
Lab 4

Lab 4 submission

yinyangcolby committed:
Lab 3 Updated So Forms Work

Forums works now.

yinyangcolby committed:
Pages updated with correct navbar

yinyangcolby committed:
Forums has bootstrap

readyR0840 commented:
Sure no problem. I will check back and fill it out later.

readyR0840 reviewed (commented):
### Checklist
For me to repost later
| **Criteria**                                                 | Check |
| ------------------------------------------------------------ | ----- |
| Does it compile and run without errors                       | [ Yes]   |
|                                                              |       |
| Do all the pages load correctly?                             | [ Yes]   |
|                                                              |       |
| Do all the links, buttons or other UI elements work correctly? | [Yes ]   |
|                                                              |       |
| Does the web app provide a good user experience (UX)?        | [Yes ]   |
|                                                              |       |
| Does the naming and structure conform to MVC conventions?<br />(Names of controllers, controller methods, views, etc.) | [Yes ]   |
|                                                              |       |
| Does the style conform to C# coding conventions?             | [ Yes]   |
|                                                              |       |
| Do the design and implementation conform to best practices?  | [No ]   There are many things that need to be addressed. I think the views are being generated with view-bag, which is not recommended (should be the model that is getting used in views).  There is also a lot going on for the controllers, when it could be done in the models, but at least it works. (for example, if you pass in the model, you could extract from the model, instead of passing in each separate value | 
|                                                              |       |
| Does the solution meet all the requirements in the lab instructions? | [ No]  The question Model is tested for 2 questions, but it does not test the actual controller methods. Try to create objects to put the controller action methods into, then test the returned model from the view. Also, you should probably test each question, to be sure. |
|                                                              |       |
| Summary Comments:                                            |     Tests were done, with Xunit testing, and packages to run them. The quiz runs and it works.   |
|                                                              |       |

yinyangcolby committed:
Unit testing and quiz added

yinyangcolby committed:
Updated xUnit version

xUnit version 2.4.1 was installed, causing an error in the tests. This issue is fixed and now all tests are working.
